Sources/Basic/Lock.swift

```diff
@@ -74,10 +96,19 @@ public final class FileLock {
             if flock(fd!, LOCK_EX) == 0 {
                 break
             }
-            // Retry if interrupted.
-            if errno == EINTR { continue }
+            if errno == EINTR {
+                if seconds != nil { 
+                    // If there is a timeout return false to indicate that
+                    // we couldn't aquire a lock.
+                    return false
```

Comment:
We aren't guaranteed this is why we were interrupted. We should use a flag to track whether or not the timer we installed actually fired, and check it here.